### PR TITLE
Definitions tab2

### PR DESCRIPTION
**Pull request by vaelliott**
Duh - first attempt to pull request was to master branch when not development.
I've made changes to the definitions tab to provide two views of definitions data - either a flextable with a few columns or the shinydashboard single indicator in a lot of detail.
The flextable will look a bit odd until the techdoc has been refreshed (there is some irregular formatting that I will fix in the google doc version). At some stage it would be nice to be able to download a png of the flextable but i was struggling to make that work so have put that on back burner for now.

### CODE REVIEW

File: server.R [hunk 1]
```diff
@@ -2317,59 +2317,200 @@ function(input, output, session) {
...
+      geo_selection <- sort(unique(c(as.character(optdata$areatype[grep(input$profile_picked,optdata$profile_domain1)]),
+                                     as.character(optdata$areatype[grep(input$profile_picked,optdata$profile_domain2)] ))))
+    } else {geo_selection <- areatype_list }
     
-    selectizeInput("indicator_defined", label = "Select indicator to see technical information for",
-                   width = "510px", choices = indic_selection, 
-                   selected = character(0), multiple=TRUE, 
-                   options = list(placeholder = "Select your indicator of interest", maxItems = 1)) 
+    selectizeInput("techdoc_geotype", label = "3. Select a geography type (optional)",
+                   width = "500px", choices = c("Show all", geo_selection),
```
Comment (jvillacampa) [inline server.R]: I would avoid using fixed widths as this messes up the layout in mobiles or different resolutions. You could use percentages or use columns.

File: test ft.R [hunk 2]
```diff
@@ -0,0 +1,95 @@
+library(shiny)
+library(dplyr)
```
Comment (jvillacampa) [inline test ft.R]: you have commited this file, but it should be deleted

File: ui.R [hunk 3]
```diff
@@ -526,106 +526,117 @@ navbarMenu("Info", icon = icon("info-circle"),
...
+                                e.g. 'Alcohol' or 'Drugs'. Profiles are further divided into topic areas to group similar indicators together. 
+                                 This page allows users to see available indicators and geographies as well as finding detailed technical information 
+                                  about how incidators are created.")),
+                             br(),
+                             #div(style = "width:60%; margin-left: 5%; min-width: 350px", # centering div
+                                 radioGroupButtons("techdoc_selection", status = "primary",
+                                                   choices = c("List of available indicators", "Detailed information about single indicator"), label= "Step 1. Select what you want to see:" ),
+                             br(),
+                             selectizeInput("profile_picked", label = "2. Select a single profile e.g. Health & wellbeing (optional)",
+                                            width = "500px",choices = profile_list_filter, selected = "Show all", multiple=FALSE),
```
Comment (jvillacampa) [inline ui.R]: relative widths instead of absolute ones

File: ui.R [hunk 4]
```diff
@@ -526,106 +526,117 @@ navbarMenu("Info", icon = icon("info-circle"),
...
+                             #conditional panel for profile summary
+                             conditionalPanel(condition = 'input.techdoc_selection == "List of available indicators"',
+                                              uiOutput("tecdoc_geographies"),
+                                              #savechart_button('download_techdoc1_png', 'Save Indicator Summary (.png)',  class = "down"),
+                                              downloadButton("download_techdoc1_csv",'Download Indicator Summary (.csv)', class = "down")),
+                             #conditional panel for single indicator
+                             conditionalPanel(condition = 'input.techdoc_selection == "Detailed information about single indicator"',
+                                              div(style="display:inline-block", 
+                                                  selectizeInput("topic_defined", label = "2b. Select a topic within a particular profile (optional)",
+                                                                 width = "500px", choices = topic_list_filter,
```
Comment (jvillacampa) [inline ui.R]: relative widths instead of absolute ones

File: ui.R [hunk 5]
```diff
@@ -526,106 +526,117 @@ navbarMenu("Info", icon = icon("info-circle"),
...
-                                   p(paste("Links to supporting information"), style="font-weight:bold; font-size: 16px; color: #2FA4E7;"),
-                                   h5(textOutput("supporting_info"))),
-                               div(class="definitionbox",
-                                   p(paste("Next update due"), style="font-weight:bold; font-size: 16px; color: #2FA4E7;"),
-                                   h5(textOutput("next_update")))
-                             ) #conditional Panel bracket 
-                      )  #column bracket
-                    )#fluidRow bracket
-           ), #Tab panel bracket             
+                      column(width = 12,
```
Comment (jvillacampa) [inline ui.R]: I think this bit can be simplified, not sure you need the column nor the div. I will do some changes here.

File: server.R [hunk 6]
```diff
@@ -2317,59 +2317,200 @@ function(input, output, session) {
...
+      else { # user selects a single geography type
+        techdoc %>%
+          subset(grepl(input$techdoc_geotype,available_geographies))}}
+  })
+  
+  ## Function to manipulate filtered data - easier for data download if manipulations done after filters
+  formatted_techdoc_data <- function(){
+    if (input$profile_picked != "Show all"){
+      techdoc_indicator_data() %>%
+        mutate(test=regexpr((names(profile_list[unname(profile_list) == input$profile_picked])), domain), #find start position of profile name in domain column
```
Comment (jvillacampa) [inline server.R]: Could test and test2 be renamed?

File: test ft.R (hunk 2 above)
Comment (vaelliott) [reply test ft.R]: removed

File: server.R (hunk 1 above)
Comment (vaelliott) [reply server.R]: converted to 100%

File: ui.R (hunk 3 above)
Comment (vaelliott) [reply ui.R]: converted to %

File: ui.R (hunk 4 above)
Comment (vaelliott) [reply ui.R]: converted to %

File: server.R (hunk 6 above)
Comment (vaelliott) [reply server.R]: done

**Pull request merged**